### ISSUE TITLE
Walmart Onn UHD

### ISSUE THREAD

**Issue by parkerlreed** [ods-releases/dopinder#2]
So I was already bootloader unlocked and ready to test this

Rebooted into real fastboot (not the fastbootd that the auto generated wiki leads you to (dtb and dtbo are not available from fastbootd))

After flashing both of the dtb file from the wiki links and attempting to load recovery via `boot` or `flash recovery` and `reboot` both leads me to the AMLogic Uboot mode

```
Bus 001 Device 026: ID 1b8e:c003 Amlogic, Inc. 
```

I can get out of that back to fastboot but no matter what I retry it will not load recovery

```
$ sudo ./update bulkcmd fastboot
AmlUsbBulkCmd[fastboot]
[AmlUsbRom]Err:return len=0 < strLenBusy 11
[AmlUsbRom]Inf:bulkInReply 
ERR: AmlUsbBulkCmd failed!
```

```
[parker@wolfcola Downloads]$ sudo fastboot flash dtb lineage-18.1-20210802-dtb-dopinder.img 
Sending 'dtb' (81 KB)                              OKAY [  0.014s]
Writing 'dtb'                                      OKAY [  0.195s]
Finished. Total time: 0.369s
[parker@wolfcola Downloads]$ sudo fastboot flash dtbo lineage-18.1-20210802-dtbo-dopinder.img
Sending 'dtbo' (2048 KB)                           OKAY [  0.113s]
Writing 'dtbo'                                     OKAY [  0.143s]
Finished. Total time: 0.408s
[parker@wolfcola Downloads]$ sudo fastboot boot /home/parker/Downloads/lineage-18.1-20210802-recovery-dopinder.img 
Sending 'boot.img' (24576 KB)                      OKAY [  1.249s]
Booting                                            OKAY [  0.000s]
Finished. Total time: 1.261s
```

(I tried flashing recovery as well)

**Comment by parkerlreed** [comment]
Wiki link https://wiki.oddsolutions.us/devices/dopinder/

Where does `dopinder` come from? The codename on the box I have lists either franlin or the full sti model number

**Comment by parkerlreed** [comment]
Tried the dtb and dtbo from the zip file itself and same result. Fastboot will not reboot to recovery and asking it to go to fastboot (`fastbootd`) leads to nothing

```
[parker@wolfcola Downloads]$ sudo fastboot flash dtb dtb.img 
Sending 'dtb' (81 KB)                              OKAY [  0.014s]
Writing 'dtb'                                      OKAY [  0.192s]
Finished. Total time: 0.366s
[parker@wolfcola Downloads]$ sudo fastboot flash dtbo dtbo.img 
Sending 'dtbo' (2048 KB)                           OKAY [  0.115s]
Writing 'dtbo'                                     OKAY [  0.142s]
Finished. Total time: 0.409s
[parker@wolfcola Downloads]$ sudo fastboot flash recovery lineage-18.1-20210802-recovery-dopinder.img 
Sending 'recovery' (24576 KB)                      OKAY [  1.256s]
Writing 'recovery'                                 OKAY [  2.163s]
Finished. Total time: 3.573s
[parker@wolfcola Downloads]$ sudo fastboot reboot recovery
Rebooting into recovery                            OKAY [  0.005s]
Finished. Total time: 3.015s
[parker@wolfcola Downloads]$ sudo fastboot reboot recovery
Rebooting into recovery                            OKAY [  0.005s]
Finished. Total time: 3.014s
```

**Comment by parkerlreed** [comment]
This is why I flashed `dtb` and `dtbo` from real fastboot

fastbootd gave me this
```
[parker@wolfcola Downloads]$ sudo fastboot flash dtb lineage-18.1-20210802-dtb-dopinder.img
[sudo] password for parker: 
Sending 'dtb' (81 KB)                              OKAY [  0.005s]
Writing 'dtb'                                      FAILED (remote: 'No such file or directory')
fastboot: error: Command failed
```

File exists in same folder. Is flashing `dtb` and `dtbo` from real fastboot what screwed me up?

**Comment by npjohnson** [comment]
I'll figure this out today

**Comment by parkerlreed** [comment]
Thank you! Interested to try out the project. 

The only thing I had different besides the unlocked bootloader to begin with was a magisk patched boot.img (rooted on the stock firmware). I don't think that would cause any issues getting to recovery, would it?

Seems the main issue currently is somehow with that recovery and dtb files, it's enough to make it panic on boot and only load the AMLogic Uboot mode. As noted I can still get back to fastboot if I send the bulkcmd. As far as I'm aware not bricked because I still get the logo.bin on HDMI in both cases.

**Comment by ods-releases** [comment]
Instructions now work

**Comment by parkerlreed** [comment]
So the custom dtb isn't required?

I'll have to test this out again. I did get back to stock dtb/o but the system refused to get past the boot animation on stock. 

Thanks.

**Comment by npjohnson** [comment]
@parkerlreed the factory image linked in the XDA thread if you run the flash-all.sh, you'll 100% boot up to stock.

No, the DTB isn't needed right now. Might change in the future, but not for now.

**Comment by parkerlreed** [comment]
This? https://forum.xda-developers.com/t/unofficial-lineageos-18-1-for-amlogic-g12-sm1-family-devices.4313743/post-85425737

I already had that URL :( I sideloaded that to make sure I was back on the right track a day ago. That is what I was referring to by being stuck on the boot animation. It was after flashing that from stock recovery sideload.

Here's to hoping I can finally boot the newer lineage recovery. I believe I tried 0802 back on the stock dtb and was still having issues (bootloader fastboot does not exit when telling it to reboot to flashed recovery or boot'ing straight to it). I'll see if 0805 is any different.

Thanks.
